### PR TITLE
Added configurable basicauth to uWSGI

### PR DESCRIPTION
**Pull request by blarghmatey**
**Added configurable basicauth to uWSGI**
Added basicauth configuration to uWSGI in order to allow for restricting
access to app in pre-production.

### CODE REVIEW

**Comment by pdpinch** [comment]
Cool. I'll give it a shot once the PR deployment builds. 

Am I reading this right that the username is MicroMasters?

**Comment by blarghmatey** [comment]
The username/password are set via the HTTP_AUTH_CREDENTIALS environment variable. MicroMasters is the authentication realm. Once the PR build is deployed I can set the credentials for verification.

File: uwsgi.ini [hunk 1]
```diff
@@ -9,3 +9,7 @@ pidfile=/tmp/micromasters-mast.pid
 vacuum=True
 enable-threads = true
 single-interpreter = true
+if-env = HTTP_AUTH_CREDENTIALS
+plugins = python,router_basicauth
+route = ^/.* basicauth:MicroMasters,%(_)
+endif =
```
Comment (pdpinch) [inline uwsgi.ini]: newline?

**Comment by pdpinch** [comment]
One nit. Otherwise this is 👍

**Pull request merged**